client/src/components/Nav/Settings.jsx

```diff
@@ -6,21 +6,22 @@ import { cn } from '~/utils/';
 import { useRecoilValue } from 'recoil';
 import { localize } from '~/localization/Translation';
 import store from '~/store';
+import { useMediaQuery } from '~/hooks';
 
 export default function Settings({ open, onOpenChange }) {
   const [confirmClear, setConfirmClear] = useState(false);
-  const [isMobile, setIsMobile] = useState(false);
+  const isMobile = useMediaQuery('(max-width: 768px)');
   const lang = useRecoilValue(store.lang);
 
   // Check if mobile dynamically and update
-  useEffect(() => {
+  useMediaQuery(() => {
```

Comment:
this is wrong, you are creating issues here. you don't need this block of code at all. line 13 does what you're trying to do in lines 17 - 24

Reply:
Is it correct now?